source/extensions/filters/http/jwt_authn/jwt_cache.cc

```diff
@@ -0,0 +1,66 @@
+#include "extensions/filters/http/jwt_authn/jwt_cache.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+// The default number of entries in JWT cache is 100.
+constexpr int kJwtCacheSize = 100;
+
+class JwtCacheImpl : public JwtCache {
+public:
+  JwtCacheImpl(bool enable_cache, int cache_size) {
+    if (enable_cache) {
+      if (cache_size == 0) {
+        jwt_cache_ =
+            std::make_unique<SimpleLRUCache<std::string, ::google::jwt_verify::Jwt>>(kJwtCacheSize);
+      } else {
+        jwt_cache_ =
+            std::make_unique<SimpleLRUCache<std::string, ::google::jwt_verify::Jwt>>(cache_size);
+      }
+    }
+  }
+
+  ~JwtCacheImpl() { jwt_cache_->clear(); }
+
+  ::google::jwt_verify::Jwt* find(const std::string& token) override {
+    if (!jwt_cache_) {
+      return nullptr;
+    }
+    ::google::jwt_verify::Jwt* found_jwt{};
+    SimpleLRUCache<std::string, ::google::jwt_verify::Jwt>::ScopedLookup lookup(jwt_cache_.get(),
+                                                                                token);
+    if (lookup.found()) {
+      found_jwt = lookup.value();
+      if (found_jwt) {
```

Comment:
I think we can ASSERT(found_jwt).  It should never be nullptr.

Reply:
added assert